### PR TITLE
Add docs for common error messages

### PR DESCRIPTION
**Pull request by chriskrycho**
We can expand this over time; for now, just the first couple items I know of!

### CODE REVIEW

File: docs/diagnosing-common-error-messages.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+```handlebars
+{{yield (component 'my-signature' someArg=true)}}
+```
+
+In that case, you can use the `ComponentLike` or `WithBoundArgs` helpers as discussed in [Contextual Components](ember/contextual-components.md):
+
+```typescript
+import Component from '@glimmer/component';
+import type { WithBoundArgs } from './some-other-component';
+import type { ComponentLike } from '@glint/template';
```
Comment (nlfurniss) [inline docs/diagnosing-common-error-messages.md:41]:
```suggestion
import type { WithBoundArgs } from '@glint/template';
```

Your example uses `WithBoundArgs` and should import it from glint

**Review by nlfurniss — CHANGES_REQUESTED** [review]

File: docs/diagnosing-common-error-messages.md (hunk 1 above)
Comment (chriskrycho) [reply docs/diagnosing-common-error-messages.md:41]: Ah! Thank you. 😅

File: docs/diagnosing-common-error-messages.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    default: [WithBoundArgs<typeof SomeOtherComponent, 'someArg'>];
+  };
+}
+
+export default class MyComponent extends Component<MySignature> {}
+
+declare module '@glint/environment-ember-loose/registry' {
+  export default interface Registry {
+    MySignature: typeof MySignature;
+    'my-signature': typeof MySignature;
```
Comment (dfreeman) [inline docs/diagnosing-common-error-messages.md:53]:
```suggestion
    MyComponent: typeof MyComponent;
    'my-component': typeof MyComponent;
```

File: docs/diagnosing-common-error-messages.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  }
+}
+```
+
+```handlebars
+{{yield (component 'my-signature' someArg=true)}}
```
Comment (dfreeman) [inline docs/diagnosing-common-error-messages.md:33]:
```suggestion
{{yield (component 'my-component' someArg=true)}}
```

File: docs/diagnosing-common-error-messages.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    default: [typeof SomeOtherComponent];
+  };
+}
+
+export default class MyComponent extends Component<MySignature> {}
+
+declare module '@glint/environment-ember-loose/registry' {
+  export default interface Registry {
+    MySignature: typeof MySignature;
+    'my-signature': typeof MySignature;
```
Comment (dfreeman) [inline docs/diagnosing-common-error-messages.md:27]:
```suggestion
    MyComponent: typeof MyComponent;
    'my-component': typeof MyComponent;
```

**Review by dfreeman — COMMENTED** [review]
Thank you @chriskrycho! This looks like a great starting point, and hopefully something we can use as input for including more helpful hints in the error messages we emit down the line.

**Pull request merged**